### PR TITLE
Add nf-test to metaphlan

### PR DESCRIPTION
**Pull request by LilyAnderssonLee**
Related to https://github.com/nf-core/taxprofiler/issues/403

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/metaphlan/metaphlan/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    test("sarscov2 - illumina single end [fastq]") {
+
+        config './general.nextflow.config'
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:true ], // meta map
+                    file( params.modules_testdata_base_path, + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true )
```
Comment (SPPearce) [inline modules/nf-core/metaphlan/metaphlan/tests/main.nf.test:34]:
```suggestion
                    file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true )
```

File: modules/nf-core/metaphlan/metaphlan/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    test("sarscov2 - illumina pair end [fastq]") {
+
+        config './general.nextflow.config'
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ], // meta map
+                    [ file( params.modules_testdata_base_path, + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true ),
```
Comment (SPPearce) [inline modules/nf-core/metaphlan/metaphlan/tests/main.nf.test:58]:
```suggestion
                    [ file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true ),
```

File: modules/nf-core/metaphlan/metaphlan/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                    file( params.modules_testdata_base_path, + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz',checkIfExists: true ) ]
```
Comment (SPPearce) [inline modules/nf-core/metaphlan/metaphlan/tests/main.nf.test:59]:
```suggestion
                    file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz',checkIfExists: true ) ]
```

File: modules/nf-core/metaphlan/metaphlan/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    test("sarscov2 - illumina single end [fasta]") {
+
+        config './general.nextflow.config'
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:true ], // meta map
+                    file( params.modules_testdata_base_path, + 'genomics/sarscov2/illumina/fasta/configs.fasta',checkIfExists: true )
```
Comment (SPPearce) [inline modules/nf-core/metaphlan/metaphlan/tests/main.nf.test:83]:
```suggestion
                    file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/configs.fasta',checkIfExists: true )
```

File: modules/nf-core/metaphlan/metaphlan/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    test("sarscov2 - illumina pair end [fastq] - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
+                    [ id:'test', single_end:false ], // meta map
+                    [ file( params.modules_testdata_base_path, + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true ),
```
Comment (SPPearce) [inline modules/nf-core/metaphlan/metaphlan/tests/main.nf.test:107]:
```suggestion
                    [ file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz',checkIfExists: true ),
```

File: modules/nf-core/metaphlan/metaphlan/tests/main.nf.test [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                    file( params.modules_testdata_base_path, + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz',checkIfExists: true )]
```
Comment (SPPearce) [inline modules/nf-core/metaphlan/metaphlan/tests/main.nf.test:108]:
```suggestion
                    file( params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz',checkIfExists: true )]
```

**Comment by jfy133** [comment]
@mashehu two of the tests are failing seemingly on self_hosted runs due to some permissions issue, any idea why that might be?

**Comment by SPPearce** [comment]
> @SPPearce Do you know how to solve James's question above?

Maybe, give me a few minutes to try something

**Comment by LilyAnderssonLee** [comment]
> > @SPPearce Do you know how to solve James's question above?
> 
> Maybe, give me a few minutes to try something

It's still failing. Any suggestions?

**Comment by SPPearce** [comment]
Afraid not, try asking on slack

**Comment by SPPearce** [comment]
Looks like that has fixed it, just an unstable file now.

**Review by SPPearce — APPROVED** [review]

**Comment by LilyAnderssonLee** [comment]
> Looks like that has fixed it, just an unstable file now.

Exactly. But the conda test is still failing.

**Comment by SPPearce** [comment]
> > Looks like that has fixed it, just an unstable file now.
> 
> Exactly. But the conda test is still failing.

Yes, but files that are not identical on conda are much easier to deal with.
Hopefully this works.

**Pull request merged**